api/client/create.go

```diff
@@ -98,17 +114,28 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 	if tag == "" {
 		tag = tags.DefaultTag
 	}
-
 	ref := registry.ParseReference(tag)
-	var trustedRef registry.Reference
 
-	if isTrusted() && !ref.HasDigest() {
-		var err error
-		trustedRef, err = cli.trustedReference(repo, ref)
-		if err != nil {
-			return nil, err
+	if pull {
+		if isTrusted() {
+			// For the trusted case, setting config.Image to the trust (notary) reference
+			// should be sufficient to check for updates.
+			if !ref.HasDigest() {
+				trustedRef, err = cli.trustedReference(repo, ref)
+				if err != nil {
+					return nil, err
+				}
+				config.Image = trustedRef.ImageName(repo)
+			}
+		} else {
+			// For the untrusted case, we perform a pull on the image before attempting create.
+			fmt.Fprintf(cli.err, "Pulling image '%s'\n", ref.ImageName(repo))
```

Comment:
It isn't needed; it was just informational to the user.  This was to give parity to the 'Pulling repository <foo>' given during a 'docker pull <foo>' command, but I don't mind removing it.